retriever/main.py

```diff
@@ -31,22 +41,37 @@
 db_port = os.getenv("POSTGRES_DATABASE_PORT")
 db_url = f"postgresql+psycopg2://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}"
 
-consumer = KafkaConsumer.from_env_vars(
-    group_id="cooler_group_id", auto_offset_reset="latest"
-)
-producer = KafkaProducer.from_env_vars()
 
-print("Connections done", file=sys.stderr)
+def get_r2r():
+    if os.getenv("OPENAI_API_TYPE") == "azure":
+        embedding_provider = AzureOpenAIEmbeddingProvider(
+            EmbeddingConfig(
+                provider="azure-openai",
+                base_model="text-embedding-3-large",
+                base_dimension=512,
+            )
+        )
+        return R2RBuilder().with_embedding_provider(provider=embedding_provider).build()
+    return R2R()
 
-logger = logging.getLogger("retriever")
+
+def get_embeddings():
+    if os.getenv("OPENAI_API_TYPE") == "azure":
+        return AzureOpenAIEmbeddings(
+            model="text-embedding-ada-002",
+            azure_deployment=os.getenv("AZURE_DEPLOYMENT_NAME"),
+            azure_endpoint=os.getenv("AZURE_OPENAI_ENDPOINT"),
+            openai_api_type=os.getenv("OPENAI_API_TYPE"),
+            openai_api_key=os.getenv("AZURE_OPENAI_API_KEY"),
+        )
+    return OpenAIEmbeddings(client="")
 
 
 def send_message(data):
-    # logger.info(f"Sending message to {flow_topic}, msg_id:", data['msg_id'])
     producer.send_message(flow_topic, data)
 
 
-async def querying_with_langchain(
+async def querying(
     session_id: str,
     turn_id: str,
     collection_name: str,
```

Comment:
collection_name is correctly taken as a parameter here. This is correct. Other places need to adopt this pattern. And not read collection_name from the ENV